mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/env python3
+
+
+import csv
+import getopt
+import os
+import subprocess
+import sys
+import math
+from prettytable import PrettyTable
+from prettytable import from_csv 
+from datetime import date
+
+# global variables.
+mlirBuildDir = './bin'
+mlirMIOpenDriver = 'mlir-miopen-driver'
+mlirROCmRunner = 'mlir-rocm-runner'
+rocprof = '/opt/rocm/bin/rocprof'
+MIOpenDriver = os.path.expanduser('~/MIOpen/build/bin/MIOpenDriver')
+benchmarkingResultFileName = 'results.stats.csv'
+configurationFileName ='../mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
+roundDigits = 2
+
+Directions = ['-F 1', '-F 4']
+DataTypes = ['conv', 'convfp16']
+Layouts = ['NCHW']
+#Layouts = ['NHWC', 'NCHW']
+
+# rocprof crashed with MIOpenDriver for these configurations on gfx908
+DisabledConfigs = ['-n 256 -c 256 -H 56 -W 56 -k 128 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1',
+                   '-n 256 -c 256 -H 56 -W 56 -k 64 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1',
+                   '-n 256 -c 64 -H 56 -W 56 -k 256 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1']
+
+# utility functions.
+def getConfigurations(fileName):
+    xdlops = False;
+    r = subprocess.run(f'/opt/rocm/bin/rocm_agent_enumerator -t GPU|grep gfx908', shell=True)
+    if r.returncode == 0:
+        xdlops = True
+
+    configFile = open(fileName, 'r')
+    Lines = configFile.readlines()
+    configs = [];
+    for direction in Directions: 
+        for datatype in DataTypes: 
+            for layout in Layouts: 
+               for line in Lines:
+                   line = line.strip()
+                   validConfig = not (xdlops and line in DisabledConfigs)
+                   if len(line) > 0 and line[0] != '#' and validConfig:
+                       oneConfig = datatype + ' ' + direction + ' -f ' + layout + ' -I '+ layout + ' -O ' + layout + ' '+ line
```

Comment:
This is better done with either `'  '.join()` or an f-string `f"{datatype} {direction} -f {layout} -I {layout} [...]"`